libs/books/feature/src/lib/book-search/book-search.component.spec.ts

```diff
@@ -28,5 +33,28 @@ describe('BookSearchComponent', () => {
 
   it('should trackByBook method return unique id', () => {
     expect(component.trackByBook(1, {id: '1'} as ReadingListBook)).toEqual('1');
-  })
+  });
+
+  it('should add book to reading list and show snack bar', () => {
+    const dispatchSpy = jest.spyOn((component as any).store, 'dispatch');
+    const removeLastAddedBookSpy = jest.spyOn((component as any), 'removeLastAddedBook');
+  
+    component.addBookToReadingList(book);
+
+    expect(dispatchSpy).toHaveBeenCalledWith(addToReadingList({book}));
+    expect(removeLastAddedBookSpy).toHaveBeenCalledWith(book);
+  });
+
+  it('should undo last action by dispatching removeFromReadingList', () => {
+    const matSnackbarOpenSpy = jest.spyOn((component as any).matSnackbar, 'open');
+    const dispatchSpy = jest.spyOn((component as any).store, 'dispatch');
+
+    (component as any).removeLastAddedBook(book);
```

Comment:
`removeLastAddedBook` and `addLastRemovedBook` are not called from a template. So, added it as `private` methods

since `removeLastAddedBook` and `addLastRemovedBook` methods are private and was getting lint error while trying to access in spec files. So, added `any`.

using square bracket notation without `any` is nice solution.